judgments/utils.py

```diff
@@ -167,3 +168,33 @@ def has_filters(query_params, exclude=["order", "per_page"]):
     be they query string, court, date, or party.
     """
     return len(set(k for (k, v) in query_params.items() if v) - set(exclude)) > 0
+
+
+def parameter_provided(params, parameter_name):
+    value = params.get(parameter_name)
+    if value and len(value) > 0:
+        return True
+
+
+def parse_parameter_as_int(params, parameter_name, default=None):
+    if parameter_provided(params, parameter_name):
+        return int(params.get(parameter_name))
+    else:
+        return default
+
+
+def parse_date_parameter(params, param_name, default_to_last=False):
+    year_param_name = f"{param_name}_year"
+    month_param_name = f"{param_name}_month"
+    day_param_name = f"{param_name}_day"
+    if parameter_provided(params, param_name):
+        return params[param_name]
+    elif parameter_provided(params, year_param_name):
```

Comment:
Why is `elif parameter_provided(params, year_param_name):` required?

Could we not just have this whole elif block as the main function block, since we return early if `param_name` is provided? 

i.e

```
def parse_date_parameter(params, param_name, default_to_last=False):
    if parameter_provided(params, param_name):
        return params[param_name]

    default_year = 1
    year = parse_parameter_as_int(params, f"{param_name}_year", default=default_year)

    default_month = 12 if default_to_last else 1
    month = parse_parameter_as_int(params, f"{param_name}_month", default=default_month)

    default_day = monthrange(year, month)[1] if default_to_last else 1
    day = parse_parameter_as_int(params, f"{param_name}_day", default=default_day)
    
    dt = datetime(year, month, day)
    return dt.strftime("%Y-%m-%d")
```

Reply:
Just need to check that doesn't choke if no year is provided for some reason, but yep, i think that'd be clearer

Reply:
aha yep, we need to explicitly handle that case or we get back "0000-1-1"

Reply:
ah fair enough! I imagine that was caught in one of the unit tests